src/uri-template.js

```diff
@@ -1,26 +1,31 @@
 import _ from 'lodash';
 
+function escapeUriTemplateVariable(variable) {
+  return encodeURIComponent(variable)
+    .replace(/[\-\.!~*'()]/g, (c) => {
```

Comment:
`encodeURIComponent` will only do half of the job and leaves `alphabetic, decimal digits, - _ . ! ~ * ' ( )` (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) unescaped so this code escapes the remaining characters so that we escape everything except alphanumeric, decimal digits and `_`.